HW4_smythe_full.py

```diff
@@ -0,0 +1,370 @@
+        Random Forest Variable importance ranking
+        Generate a table with all the variables and their rankings
+I'm going to grade this by giving it some random dataset
+and seeing if it outputs everything
+Desire: html or excel based rankings report with links to the plots
+
+    Recommendations for plots
+    Predictor / Response type Dependant
+        Response: Boolean / Categorical
+            Predictor: Boolean / Categorical
+                Heatplot
+            Predictor: Continous
+                Violin plot on predictor grouped by response
+                Distribution plot on predictor grouped by response
+        Response: Continuous
+            Predictor: Boolean / Categorical
+                Violin plot on response grouped by predictor
+                Distribution plot on response grouped by predictor
+            Predictor: Continuous
+                Scatter plot with trendline
+"""
+
+import sys
+from datetime import datetime
+
+import numpy as np
+import pandas as pd
+import plotly.express as px
+import statsmodels.api as sm
+from sklearn import datasets
+from sklearn.linear_model import LogisticRegression
+from sklearn.utils.multiclass import type_of_target as tot
+
+
+# PLOT Functions
+def plot_cont_cont(to_plot):
+    # Scatter plot with trendline
+    # from class notes Week6
+    user_db = to_plot
+    X = user_db.data
+    y = user_db.target
+    con_con_file_list = []
+    for idx, column in enumerate(X.T):
+        feature_name = user_db.feature_names[idx]
+        feature_name = feature_name.replace("/", "-")
+        predictor = sm.add_constant(column)
+        linear_regression_model = sm.OLS(y, predictor)
+        linear_regression_model_fitted = linear_regression_model.fit()
+        print(f"Variable: {feature_name}")
+        print(linear_regression_model_fitted.summary())
+
+        # Get the stats
+        t_value = round(linear_regression_model_fitted.tvalues[1], 6)
+        p_value = "{:.6e}".format(linear_regression_model_fitted.pvalues[1])
+
+        # Plot the figure
+        fig = px.scatter(x=column, y=y, trendline="ols")
+        fig.update_layout(
+            title=f"Variable: {feature_name}: (t-value={t_value}) (p-value={p_value})",
+            xaxis_title=f"Variable: {feature_name}",
+            yaxis_title="y",
+        )
+        file_name = f"./HW4_{feature_name}_lin_plot.html"
+        fig.write_html(file=file_name, include_plotlyjs="cdn")
+        con_con_file_list += [file_name]
+    return con_con_file_list
+
+
+def plot_cont_cat(to_plot):
+    # Violin plot on predictor grouped by response
+    user_db = to_plot
+    X = user_db.data
+    y = user_db.target
+    con_cat_file_list = []
+    #
+    for idx, column in enumerate(X.T):
+        feature_name = user_db.feature_names[idx]
+        feature_name = feature_name.replace("/", "-")
+        predictor = sm.add_constant(column)
+        #
+        logistic_regression_model = LogisticRegression(random_state=4622).fit(

```

Comment:
I believe we were supposed to use statsmodels.api. There's an option for lin_regr_model and log_regr_model. The reason I say this is because we were supposed to get stats (p and t values) on both categorical and continuous data. So we had to use statsmodels.api